client/pkid_client.go

```diff
@@ -0,0 +1,253 @@
+package client
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	sodium "github.com/gokillers/libsodium-go/cryptosign"
+)
+
+type PkidClient struct {
+	client     http.Client
+	port       int
+	privateKey []byte
+	publicKey  []byte
+}
+
+// create a new instance from the pkid client
+func NewPkidClient(privateKey []byte, publicKey []byte, port int) PkidClient {
+	client := http.Client{Timeout: 5 * time.Second}
+
+	return PkidClient{
+		client:     client,
+		port:       port,
+		privateKey: privateKey,
+		publicKey:  publicKey,
+	}
+
+}
+
+// generate a private key and public key for the client
+func GenerateKeyPair() ([]byte, []byte) {
+	privateKey, publicKey, _ := sodium.CryptoSignKeyPair()
+	return privateKey, publicKey
+}
+
+// generate a private key and public key for the client using TF login seed
+func GenerateKeyPairUsingSeed(seed string) ([]byte, []byte, error) {
+	decodedSeed, err := base64.StdEncoding.DecodeString(seed)
+	if err != nil {
+		return []byte{}, []byte{}, err
+	}
+	privateKey, publicKey, _ := sodium.CryptoSignSeedKeyPair(decodedSeed)
+	return privateKey, publicKey, nil
+}
+
+// set a new value for a key inside a project
+func (pc *PkidClient) Set(project string, key string, value string, willEncrypt bool) error {
+
+	if willEncrypt {
+		decryptedValue, err := encrypt(value, pc.publicKey)
+		if err != nil {
+			return errors.New("encryption failed with error: " + fmt.Sprint(err))
```

Comment:
you can use error wrapping https://go.dev/blog/go1.13-errors to get around this one